code/game/objects/items/RPD.dm

```diff
@@ -264,6 +264,14 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 		return TRUE
 	return ..()
 
+/obj/item/pipe_dispenser/pre_attack_secondary(obj/machinery/atmospherics/target, mob/user, params)
+	if(!istype(target, /obj/machinery/atmospherics))
+		return ..()
+	if(target.pipe_color)
+		paint_color = GLOB.pipe_color_name[target.pipe_color]
```

Comment:
you could add the sparks here to signal that something happened (or even a message to the user)

Reply:
My main concern with sparks is that it would be too spammy. I've added a chat message to the user.